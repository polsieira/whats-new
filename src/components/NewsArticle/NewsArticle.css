.NewsArticle {
  width: 33%;
}

.image--news {
  width: 80%;
}