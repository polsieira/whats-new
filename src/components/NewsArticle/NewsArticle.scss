.NewsArticle {
  background: #ffffff;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 10px;
  width: 100%;

  h3 {
    padding: 8px;
  }

  p {
    padding: 10px;
  }

  a {
    align-items: center;
    background:  #B56AF0;
    color: #ffffff;
    display: flex;
    justify-content: space-around;
    padding: 5px;
    text-decoration: none;
    
    &:hover {
      background:  #FFAE02;
    }
  }
}

.image--news {
  object-fit: cover;
  height: 300px;
  width: 100%;
}

