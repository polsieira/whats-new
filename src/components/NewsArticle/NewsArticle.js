import React from 'react';
import './NewsArticle.scss';

const NewsArticle = ({ headline, description, url, img }) => {
  return (
    <div className="NewsArticle">
      <img className="image--news" src={img} alt="News article" />
      <h3>{headline}</h3>
      <p>{description}</p>
      <a href={url}>Link To Article</a>
    </div>
  )
}

export default NewsArticle;