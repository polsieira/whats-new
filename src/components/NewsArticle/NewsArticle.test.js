import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      img='React Logo With Cat Hanging Off It'
      headline='Student Learns React'
      description='He failed to learn React. The End.'
      url='www.CatReact.com' />);

    expect(wrapper).toMatchSnapshot();
  });
});