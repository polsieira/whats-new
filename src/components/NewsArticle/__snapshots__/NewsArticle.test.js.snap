// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<div
  className="NewsArticle"
>
  <img
    alt="News article"
    className="image--news"
    src="React Logo With Cat Hanging Off It"
  />
  <h3>
    Student Learns React
  </h3>
  <p>
    He failed to learn React. The End.
  </p>
  <a
    href="www.CatReact.com"
  >
    Link To Article 
    <IoIosArrowRoundForward
      size={30}
    />
  </a>
</div>
`;
